[SITE-786] Drupal PAPC - Updates Surrogate-key header limit (#9584)
pantheon-systems/documentation: source/releasenotes/2025-06-19-drupal-papc-surrogate-key-limit-update.md

---

> title: Pantheon Advanced Page Cache Surrogate Key Limit Update

Reviewer: ```suggestion
title: Pantheon Advanced Page Cache Drupal Module Surrogate Key Limit Update
```

published_date: "2025-06-19"
categories: [drupal, modules]
---

> This update on the [Pantheon Advanced Page Cache](https://www.drupal.org/project/pantheon_advanced_page_cache) Drupal module focuses on improving the functionality of the surrogate key header limit in your local development environment.

Reviewer: ```suggestion
Version 2.3.3 of the [Pantheon Advanced Page Cache](https://www.drupal.org/project/pantheon_advanced_page_cache) Drupal module adds support for overriding the surrogate key header size limit to better support local development environments.
```

**Key Update**
Surrogate Key Header Limit Adjustment: To prevent header breaking and enhance the ease of work in your local environment, you now have the ability to override the surrogate key header limit. 

> By default, this limit is set to 0 bytes. You can adjust this limit by adding the following line to your site's settings.php file:

Reviewer: > this limit is set to 0 bytes

Does 0 = unlimited in this case? Or are we not outputting any surrogate key header locally by default?

Author: It means "unlimited" and not "disabled." It's still being output without a limit, as some developers might want to test the headers locally.

> By default, this limit is set to 0 bytes. You can adjust this limit by adding the following line to your site's settings.php file:

Reviewer: ```suggestion
By default, this limit is set to 0 (unlimited). You can adjust this limit by adding the following line to your site's settings.php file:
```

Reviewer: Actually... does 0 = 25000? Or setting to 0 goes beyond 25000?

Author: $limit defaults to 25,000 if not configured.
Sets $limit at 25,000 if the user tries to set it under zero or above 25000
Caps $limit to 25,000
